src/Kernel32/storebanned/Kernel32.cs

```diff
@@ -3026,5 +3045,166 @@ public static partial class Kernel32
+        /// thread gets a default security descriptor.The ACLs in the default security descriptor for a thread
+        /// come from the primary token of the creator.
+        /// </param>
+        /// <param name="dwStackSize">The initial size of the stack, in bytes. The system rounds
+        /// this value to the nearest page. If this parameter is 0 (<see cref="SIZE_T.Zero"/>), the new thread uses the default size
+        /// for the executable.</param>
+        /// <param name="lpStartAddress">A pointer to the application-defined function to be executed
+        /// by the thread. This pointer represents the starting address of the thread. For more
+        /// information on the thread function, <see cref="THREAD_START_ROUTINE"/>.</param>
+        /// <param name="lpParameter">A pointer to a variable to be passed to the thread.</param>
+        /// <param name="dwCreationFlags">
+        /// The flags that control the creation of the thread.
+        /// <see cref="CreateProcessFlags.None"/>, <see cref="CreateProcessFlags.CREATE_SUSPENDED"/>
+        /// and <see cref="CreateProcessFlags.STACK_SIZE_PARAM_IS_A_RESERVATION"/> are the only valid
+        /// values for this parameter.
+        /// </param>
+        /// <param name="lpThreadId">A pointer to a variable that receives the thread identifier. If this
+        /// parameter is null, the thread identifier is not returned.</param>
+        /// <returns>
+        /// If the function succeeds, the return value is a handle to the new thread.
+        /// If the function fails, the return value is null.To get extended error information, call
+        /// <see cref="GetLastError"/>.
+        /// </returns>
+        /// <remarks>
+        /// Note that <see cref="CreateThread(SECURITY_ATTRIBUTES*, SIZE_T, THREAD_START_ROUTINE, IntPtr, CreateProcessFlags, int*)"/>
+        /// may succeed even if <paramref name="lpStartAddress"/> points to data, code, or is not accessible.
+        /// If the start address is invalid when the thread runs, an exception occurs, and the thread terminates.
+        /// Thread termination due to a invalid start address is handled as an error exit for the thread's process. This behavior
+        /// is similar to the asynchronous nature of <see cref="CreateProcessAsUser(IntPtr, string, string, SECURITY_ATTRIBUTES*, SECURITY_ATTRIBUTES*, bool, CreateProcessFlags, void*, string, ref STARTUPINFO, out PROCESS_INFORMATION)"/>,
+        /// where the process is created even if it refers to invalid or missing dynamic-link libraries (DLLs).
+        /// </remarks>
+        [DllImport(api_ms_win_core_processthreads_l1_1_1, SetLastError = true)]
+        public static extern unsafe SafeThreadHandle CreateThread(
+            [Friendly(FriendlyFlags.In)] SECURITY_ATTRIBUTES* lpThreadAttributes,
+            SIZE_T dwStackSize,
+            THREAD_START_ROUTINE lpStartAddress,
+            IntPtr lpParameter,
+            CreateProcessFlags dwCreationFlags,
```

Comment:
Is it really appropriate to reuse `CreateProcessFlags` for the `CreateThread` method?

Reply:
They share one enum value - CREATE_SUSPENDED. They also share the sentinel value `None`. 

That left me with a choice - expand CreateProcessFlags by 1 enum value (STACK_SIZE_PARAM_IS_A_RESERVATION), or create a new enum and add STACK_SIZE_PARAM_IS_A_RESERVATION, CREATE_SUSPENDED, and None to it. 

I don’t like the name `CreateProcessFlags` being used here, but these flags being in the same family (is the same enum) is reasonable from a Win32 standpoint (Process, threads, fibers go together). 

OTOH I don’t think thread creation fns would undergo fundamental breaking changes, and we could probably create `CreateThreadFlags` enum as a clean way if calling these methods. 

While pit-of-success design would normally compel me to instantiate a new enum type, calling CreateThread from managed code is an uncommon/advanced pattern and developers would likely know what they are doing. 

I could go either way. Would you rather I added a new enum?

Reply:
Ya, I'd rather a new enum. Given `CreateProcessFlags` has so many flags that don't apply her, the overlap is small.